app/shared/ui-foundation/src/commonMain/kotlin/ui/foundation/lists/PaginatedListState.kt

```diff
@@ -0,0 +1,268 @@
+
+/**
+ * 分组数据接口，定义分组的基本属性
+ */
+interface GroupData {
+    val title: String
+}
+
+/**
+ * 可分页列表的分组配置
+ */
+interface PaginatedGroupConfig<T> {
+    /**
+     * 每个分组的最大条目数
+     */
+    val itemsPerGroup: Int
+
+    /**
+     * 检查某个条目是否属于指定的播放状态
+     */
+    fun isPlaying(item: T): Boolean = false
+
+    /**
+     * 生成指定分组的标题
+     */
+    fun generateGroupTitle(groupIndex: Int, itemsInGroup: List<T>): String
+}
+
+/**
+ * 默认的分组配置实现
+ */
+class DefaultPaginatedGroupConfig<T>(
+    override val itemsPerGroup: Int,
+    private val playingChecker: ((T) -> Boolean)? = null,
+    private val titleGenerator: (Int, List<T>) -> String,
+) : PaginatedGroupConfig<T> {
+    override fun isPlaying(item: T): Boolean = playingChecker?.invoke(item) ?: false
+    override fun generateGroupTitle(groupIndex: Int, itemsInGroup: List<T>): String =
+        titleGenerator(groupIndex, itemsInGroup)
+}
+
+/**
+ * 分页列表分组数据
+ */
+@Stable
+data class PaginatedGroup<T>(
+    override val title: String,
+    val items: List<T>,
+    val startIndex: Int,
+    val groupIndex: Int,
+) : GroupData
+
+/**
+ * 分页列表状态管理类
+ *
+ * 负责管理分页列表的分组、导航、定位等核心逻辑
+ */
+@Stable
+class PaginatedListState<T>(
+    private val items: List<T>,
+    private val config: PaginatedGroupConfig<T>,
+    initialPlayingIndex: Int = -1,
+) {
+    /**
+     * 分组后的数据
+     */
+    val groups: List<PaginatedGroup<T>> by derivedStateOf {
+        items.chunked(config.itemsPerGroup).mapIndexed { groupIndex, chunk ->
+            val startItemIndex = groupIndex * config.itemsPerGroup
+            PaginatedGroup(
+                title = config.generateGroupTitle(groupIndex, chunk),
+                items = chunk,
+                startIndex = startItemIndex,
+                groupIndex = groupIndex,
+            )
+        }
+    }
+
+    /**
+     * 当前播放条目的索引
+     */
+    val playingIndex: Int by derivedStateOf {
+        items.indexOfFirst { config.isPlaying(it) }
+    }
+
+    /**
+     * 当前分组索引
+     */
+    var currentGroupIndex by mutableStateOf(
+        if (initialPlayingIndex >= 0) initialPlayingIndex / config.itemsPerGroup else 0,
+    )
+
+    /**
+     * 是否显示分组选择器
+     */
+    var showGroupSelector by mutableStateOf(false)
+
+    /**
+     * 是否处于初始定位状态
+     */
+    var isInitialPositioning by mutableStateOf(true)
+
+    /**
+     * 每个分组在列表中的起始索引
+     */
+    val groupStartIndices: List<Int> by derivedStateOf {
+        var index = 0
+        groups.map { group ->
+            val startIndex = index
+            index++ // group header
+            index += group.items.size // items
+            startIndex
+        }
+    }
+
+    /**
+     * 总条目数
+     */
+    val totalItemsCount: Int get() = items.size
+
+    /**
+     * 总分组数
+     */
+    val totalGroupsCount: Int get() = groups.size
+
+    /**
+     * 当前分组数据
+     */
+    val currentGroup: PaginatedGroup<T>?
+        get() = groups.getOrNull(currentGroupIndex)
+
+    /**
+     * 是否可以导航到上一组
+     */
+    val canNavigateToPrevious: Boolean
```

Comment:
```suggestion
    val canNavigateToPreviousGroup: Boolean
```

这个 class 里涉及 item 和 group, 最好区分一下概念